README.md

```diff
@@ -1,29 +1,103 @@
-## to setup, create a virtual env and activate
-# python -m venv dw_venv
-#  .\dw_venv\Scripts\activate
+# 📦 DreamWeaver.AI Streamlit App Starter Kit
 
-## install requirements
-# python -m pip install -r requirements.txt
+An AI-powered sleep app designed to help you analyze and improve your sleep quality through advanced AI recommendations.
 
-## to run:
-# python -m streamlit run app.py
+## Setup Instructions
 
-# 📦 DreamWeaver.AI Streamlit App Starter Kit 
+### 1. Create and Activate a Virtual Environment
 
-An AI sleep app.
+First, create a virtual environment:
 
-## Demo App
+```bash
+python -m venv dw_venv
+```
+
+Activate the environment:
+
+- **Windows:**
+
+    ```bash
+    .\dw_venv\Scripts\activate
+    ```
+
+- **macOS/Linux:**
+
+    ```bash
+    source dw_venv/bin/activate
+    ```
+
+### 2. Install Requirements
+
+Install the necessary dependencies:
+
+```bash
+python -m pip install -r requirements.txt
+```
+
+### 3. Run the Application
+
+To start the application, run the following command:
+
+```bash
+python -m streamlit run app.py
+```
+
+## Secrets Configuration
+
+To keep your API keys and other sensitive information secure, add them to the `secrets.toml` file.
+
+**Example `secrets.toml` Configuration:**
+
+```toml
+[general]
+GOOGLE_API_KEY = "your_api_key"
+```
+
```

Comment:
## To get a google Gemini API key
Go to https://aistudio.google.com/app/apikey, sign in with your google account and click 'Get API Key' -> 'Generate API Key'